x-pack/plugins/reporting/server/export_types/png/lib/generate_png.ts

```diff
@@ -40,20 +49,24 @@ export async function generatePngObservableFactory(reporting: ReportingCore) {
       layout,
       browserTimezone,
     }).pipe(
-      map((results: ScreenshotResults[]) => {
-        if (apmScreenshots) apmScreenshots.end();
-        if (apmTrans) apmTrans.end();
+      tap(([{ screenshots }]) => {
+        const [{ base64EncodedData }] = screenshots;
+        const byteLength = getBase64DecodedSize(base64EncodedData);
 
-        return {
-          base64: results[0].screenshots[0].base64EncodedData,
-          warnings: results.reduce((found, current) => {
-            if (current.error) {
-              found.push(current.error.message);
-            }
-            return found;
-          }, [] as string[]),
-        };
-      })
+        logger.debug(`PNG buffer byte length: ${byteLength}`);
+        apmTrans?.setLabel('byte_length', byteLength, false);
```

Comment:
For PNG jobs, it looks like this metric is working, but not the `cpu` and `memory` ones. 

![image](https://user-images.githubusercontent.com/908371/118048388-13175e00-b331-11eb-9acf-1dfe21c3b51f.png)

@dokmic I'm OK with having the PR merged as long as the PDF metrics are collected.